File: ios/Plugin/MapboxCapacitorPluginPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(MapboxCapacitorPluginPlugin, "MapboxCapacitorPlugin",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showHelloWorld, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showHelloWorld2, CAPPluginReturnPromise);
)

File: ios/Plugin/MapboxCapacitorPluginPlugin.swift
import Foundation
import Capacitor

@objc(MapboxCapacitorPluginPlugin)
public class MapboxCapacitorPluginPlugin: CAPPlugin {

    @objc func showHelloWorld(_ call: CAPPluginCall) {
            DispatchQueue.main.async {
                let helloWorldVC = MapboxCapacitorPluginViewController()
                let navigationController = UINavigationController(rootViewController: helloWorldVC)
                
                if let viewController = self.bridge?.viewController {
                    viewController.present(navigationController, animated: true, completion: nil)
                }
            }
            call.resolve()
        }
}
